akka/src/main/scala/chatapp/Chat.scala

```diff
@@ -16,81 +16,79 @@ object Action extends Enumeration {
 import Action._
 
 class BehaviorFactory(compute: Int, post: Int, leave: Int, invite: Int) {
-  def apply(dice: DiceRoll): Action = {
-    val pick = dice()
+  val _compute: Int = compute
+  val _post: Int = _compute + post
+  val _leave: Int = _post + leave
+  val _invite: Int = _leave + invite
+
+  def apply(pick: Int): Action = {
     var action: Action = None
 
-    if (pick < compute) {
+    if (pick < _compute) {
       action = Compute
-    } else if (pick < post) {
+    } else if (pick < _post) {
       action = Post
-    } else if (pick < leave) {
+    } else if (pick < _leave) {
       action = Leave
-    } else if (pick < invite) {
+    } else if (pick < _invite) {
       action = Invite
     }
-
     action
   }
 }
 
 
 sealed trait ChatMsg
-final case class ChatPost(payload: Array[Byte], accumulator: ActorRef[Accumulator]) extends ChatMsg
-final case class ChatJoin(client: ActorRef[Client], accumulator: ActorRef[Accumulator]) extends ChatMsg
-final case class ChatLeave(client: ActorRef[Client], didLogout: Boolean, accumulator: ActorRef[Accumulator]) extends ChatMsg
+final case class ChatPost(payload: Array[Byte], accumulator: ActorRef[AccMsg]) extends ChatMsg
```

Comment:
Hm, am I misunderstanding what an ActorRef's type should be?
It's the message type not the type of the actor object?

Is there any way for me to get this understanding from the docs?
Looking at https://doc.akka.io/api/akka/current/akka/actor/typed/ActorRef.html
T is not explicitly documented.

Though, I see it used in some method signatures to type messages.
So, I guess your change is correct. Still strange to me.

Reply:
I used the reference docs, see https://doc.akka.io/docs/akka/current/typed/actors.html